pkg/controllers/namespace/subnet_share_controller.go

```diff
@@ -196,7 +196,12 @@ func (r *NamespaceReconciler) deleteUnusedSharedSubnets(ctx context.Context, ns
 			} else {
 				log.Info("Deleted Subnet CR for shared Subnet",
 					"Namespace", ns, "Name", subnet.Name, "AssociatedResource", associatedResource)
+				r.StatusUpdater.DeleteSuccess(client.ObjectKey{Namespace: ns, Name: subnet.Name}, subnet)
 			}
+		} else {
+			err := fmt.Errorf("subnet CR %s/%s is still referenced and cannot be deleted", subnet.Namespace, subnet.Name)
+			log.Error(nil, "Cannot delete Subnet CR for shared subnet because it is still referenced")
+			return err
```

Comment:
Add the error to a list so that the failure to delete a Subnet will not block the deletion of other Subnets?

Reply:
In checkSubnetReferences, just use the k8s client to get CR. If this basic usage goes wrong, it means that k8s is unavailable and the whole system will have problems so that we can return early.